src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/LayoutUtils.java

```diff
@@ -61,21 +61,26 @@ public static String getName(Cursor cursor) {
 
     public static MemoryLayout getLayout(Type t) {
         switch(t.kind()) {
-            case Char_S:
-            case Char_U:
             case UChar:
+            case Char_U:
+                return C_UCHAR;
```

Comment:
Could use

```java
            case UChar, Char_U:
                return C_UCHAR;
```

Here and elsewhere in the same switch.

Or maybe it's nice to switch the whole switch to a switch-expression using `->`, since then you can remove all the `return`s?

Reply:
Agree on combining cases. But there is control flow (throw with couple of more statements). I'd prefer minimal change for this fix.